### PR TITLE
Fixes for SoA NEURON data structures

### PR DESCRIPTION
**Pull request by olupton**
Fixes needed for compatibility with neuronsimulator/nrn#2027.

### CODE REVIEW

File: modeldb/modeldb-run.yaml [hunk 1]
```diff
@@ -1219,6 +1229,7 @@
     comment: Tries to open bac6.ses, which does not exist
     run: null
 267384:
+    github: 'pull/2'
```
Comment (olupton) [inline modeldb/modeldb-run.yaml:1232]: This one should be merged imminently; it slipped through the cracks when the others went in.

File: modeldb/modeldb-run.yaml [hunk 2]
```diff
@@ -340,13 +340,26 @@
...
+      repl: ''
+    - pattern: '^\s+(finitialize|init|run|stdinit)\((|-65)\)'
+      repl: ''
+    - pattern: '^\s+\^'
+      repl: ''
     run:
     - ringperf()
     - run()
-    - verify_graph_()
+    # - verify_graph_()
```
Comment (alexsavulescu) [inline modeldb/modeldb-run.yaml:362]: artifact?

File: modeldb/modeldb-run.yaml [hunk 3]
```diff
@@ -692,15 +705,12 @@
...
-    curate_patterns:
-    - pattern: '^([0-9.\-e]+) kilocycles$'
-      repl: '%some_count% kilocycles'
     run:
     - load_file("batch_.hoc")
     - verify_graph_()
     script:
     # 424242: this is just avoiding a timing-related number being printed
-    - sed -i'.bak' -e 's/batch_flag=0/batch_flag=1/g;s/tstop = 1e3/tstop = 20/g;s#return tti/1e3#return 424242#g' batch_.hoc
+    - sed -i.bak  -e 's/batch_flag=0/batch_flag=1/g;s/tstop = 1e3/tstop = 20/g;s#return tti/1e3#return 424242#g;s#print tti/.*#print "%some_time%"#g' batch_.hoc
```
Comment (alexsavulescu) [inline modeldb/modeldb-run.yaml:713]: I remember the only pattern that was portable on macOS was like
```suggestion
    - sed -i'.bak' -e 's/batch_flag=0/batch_flag=1/g;s/tstop = 1e3/tstop = 20/g;s#return tti/1e3#return 424242#g;s#print tti/.*#print "%some_time%"#g' batch_.hoc
```

File: modeldb/modeldb-run.yaml (hunk 2 above)
Comment (olupton) [reply modeldb/modeldb-run.yaml:362]: No, intentional: see the comment above.

**Review by alexsavulescu — APPROVED** [review]

**Pull request merged**